multivariate_grouped_data_plot.Rmd

```diff
@@ -0,0 +1,825 @@
+  geom_point(data = df.summary, size = 2)
+# (2) Bar plots of means + individual jitter points + errors
+ggplot(df, aes(dose, len)) +
+  geom_bar(stat = "identity", data = df.summary,
+           fill = NA, color = "black") +
+  geom_jitter( position = position_jitter(0.2),
+               color = "black") + 
+  geom_errorbar(
+    aes(ymin = len-sd, ymax = len+sd),
+    data = df.summary, width = 0.2) 
+```
+
+#### 3.7.2 多个组的均值/中位数图。 
+
+一个连续变量（len）和两个分组变量（dose，supp）的情况。
+
+(1) 计算按dose和supp分组的len的摘要统计量。
+
+```{r}
+library(dplyr)
+df.summary2 <- df %>%
+  group_by(dose, supp) %>%
+  summarise(
+    sd = sd(len),
+    len = mean(len)
+  )
+df.summary2
+```
+
+(2) 为多个组创建误差图。
+
+- 按组着色的点范围（supp）
+
+- 标准误差线+按组着色的平均值（supp）
+
+```{r}
+# (1) Pointrange: Vertical line with point in the middle
+ggplot(df.summary2, aes(dose, len)) +
+  geom_pointrange(
+    aes(ymin = len-sd, ymax = len+sd, color = supp),
+    position = position_dodge(0.3)
+    )+
+  scale_color_manual(values = c("#00AFBB", "#E7B800"))
+# (2) Standard error bars
+ggplot(df.summary2, aes(dose, len)) +
+  geom_errorbar(
+    aes(ymin = len-sd, ymax = len+sd, color = supp),
+    position = position_dodge(0.3), width = 0.2
+    )+
+  geom_point(aes(color = supp), position = position_dodge(0.3)) +
+  scale_color_manual(values = c("#00AFBB", "#E7B800"))
+```
+
+(3) 为多个组创建简单的线/条形图。
+
+- 线图：按组更改线型（Supp）
+
+- 条形图：按组更改填充颜色（Supp）
+
+```{r}
+# (1) Line plot + error bars
+ggplot(df.summary2, aes(dose, len)) +
+  geom_line(aes(linetype = supp, group = supp))+
+  geom_point()+
+  geom_errorbar(
+    aes(ymin = len-sd, ymax = len+sd, group = supp),
+     width = 0.2
+    )
+# (2) Bar plots + upper error bars.
+ggplot(df.summary2, aes(dose, len)) +
+  geom_bar(aes(fill = supp), stat = "identity",
+           position = position_dodge(0.8), width = 0.7)+
+  geom_errorbar(
+    aes(ymin = len, ymax = len+sd, group = supp),
+    width = 0.2, position = position_dodge(0.8)
+    )+
+  scale_fill_manual(values = c("grey80", "grey30"))
+```
+
+(4) 创建多个组的平均值+/-sd的图。 
+
+使用ggpubr软件包，该软件包将自动计算摘要统计信息并创建图形。
+
+```{r}
+library(ggpubr)
+# Create line plots of means
+ggline(ToothGrowth, x = "dose", y = "len", 
+       add = c("mean_sd", "jitter"),
+       color = "supp", palette = c("#00AFBB", "#E7B800"))
+# Create bar plots of means
+ggbarplot(ToothGrowth, x = "dose", y = "len", 
+          add = c("mean_se", "jitter"),
+          color = "supp", palette = c("#00AFBB", "#E7B800"),
+          position = position_dodge(0.8))
+```
+
+(5) 使用标准ggplot2来重现上面的线图.
+
+```{r}
+# Create line plots
+ggplot(df, aes(dose, len)) +
+  geom_jitter(
+    aes(color = supp),
+    position = position_jitter(0.2)
+    ) + 
+  geom_line(
+    aes(group = supp, color = supp),
+    data = df.summary2
+    ) +
+  geom_errorbar(
+    aes(ymin = len-sd, ymax = len+sd, color = supp),
+    data = df.summary2, width = 0.2
+    )+
+  scale_color_manual(values = c("#00AFBB", "#E7B800"))
+```
+
+
+   
+   
+   
+
+## 4 总结
+
+### 4.1 可视化已分组的连续变量的分布
```

Comment:
Hashtag headers should not be followed by numbers since the hashtags will create numbered subheadings.